academy/cosmjs-checkers-sections/server-side.md

```diff
@@ -628,27 +668,30 @@ const handleEventCreate = async (event: StringEvent): Promise<void> => {
 
 ## Handle one reject event
 
-`handleEventReject` is the counterpart of `handleEventCreate`, as it removes games from the system.
+`handleEventReject` is the counterpart of `handleEventCreate`, as it soft-removes games from the system.
 
-```typescript [https://github.com/cosmos/academy-checkers-ui/blob/b030b2f/src/server/indexer.ts#L160-L172]
+```typescript [https://github.com/cosmos/academy-checkers-ui/blob/server-indexing/src/server/indexer.ts#L164-L177]
 const handleEventReject = async (event: StringEvent): Promise<void> => {
-    const rejectedId: string | undefined = getAttributeValueByKey(event.attributes, "IdValue")
-    if (!rejectedId) throw new Error(`Reject event missing rejectedId`)
+    const rejectedId: string | undefined = getAttributeValueByKey(event.attributes, "game-index")
+    if (!rejectedId) throw new Error(`Reject event missing game-index`)
     const blackAddress: string | undefined = db.games[rejectedId]?.blackAddress
     const redAddress: string | undefined = db.games[rejectedId]?.redAddress
+    console.log(`Reject game: ${rejectedId}, black: ${blackAddress}, red: ${redAddress}`)
     const blackGames: string[] = db.players[blackAddress]?.gameIds ?? []
     const redGames: string[] = db.players[redAddress]?.gameIds ?? []
-    console.log(`Reject game: ${rejectedId}, black: ${blackAddress}, red: ${redAddress}`)
     const indexInBlack: number = blackGames.indexOf(rejectedId)
     if (0 <= indexInBlack) blackGames.splice(indexInBlack, 1)
     const indexInRed: number = redGames.indexOf(rejectedId)
     if (0 <= indexInRed) redGames.splice(indexInRed, 1)
+    if (db.games[rejectedId]) db.games[rejectedId].deleted = true
 }
 ```
 
 <HighlightBox type="note">
 
 * `handleEventReject` keeps the game information in the `db`. This is a debatable choice, but helps the `patch` function identify old games that once existed as opposed to games that never existed.
+* It marks the game as `deleted: true`, though.
+* And it removes the from both players' list of games.
```

Comment:
I have corrected this sentence to read "And it removes the ***id*** from both players' list of games", to conform with the example from line 726